unikernel.ml

```diff
@@ -973,62 +995,189 @@ struct
   let unikernel_info_one albatross store name reqd ~json_dict:_
       (user : User_model.user) =
     (* TODO use uuid in the future *)
-    (Albatross.query albatross ~domain:user.name ~name
-       (`Unikernel_cmd `Unikernel_info)
-     >|= function
-     | Error msg ->
-         Logs.err (fun m ->
-             m "error while communicating with albatross: %s" msg);
-         []
-     | Ok (_hdr, `Success (`Unikernel_info unikernel)) -> unikernel
-     | Ok reply ->
-         Logs.err (fun m ->
-             m "expected a unikernel info reply, received %a"
-               (Vmm_commands.pp_wire ~verbose:false)
-               reply);
-         [])
-    >>= fun unikernels ->
-    if List.length unikernels > 0 then
-      (Albatross.query_console ~domain:user.name albatross ~name >|= function
-       | Error err ->
-           Logs.warn (fun m -> m "error querying console of albatross: %s" err);
-           []
-       | Ok (_, console_output) -> console_output)
-      >>= fun console_output ->
-      let now = Ptime.v (P.now_d_ps ()) in
-      generate_csrf_token store user now reqd >>= function
-      | Ok csrf ->
-          Lwt.return
-            (reply reqd ~content_type:"text/html"
-               (Dashboard.dashboard_layout ~csrf user
-                  ~content:
-                    (Unikernel_single.unikernel_single_layout
-                       (List.hd unikernels) now console_output)
-                  ~icon:"/images/robur.png" ())
-               ~header_list:[ ("X-MOLLY-CSRF", csrf) ]
-               `OK)
-      | Error err ->
-          Lwt.return
-            (reply reqd ~content_type:"text/html"
-               (Guest_layout.guest_layout ~page_title:"500 | Mollymawk"
-                  ~content:(Error_page.error_layout err)
-                  ~icon:"/images/robur.png" ())
-               `Internal_server_error)
-    else
-      let error =
-        {
-          Utils.Status.code = 500;
-          title = "An error occured";
-          success = false;
-          data = `String "Error while fetching unikernel.";
-        }
-      in
-      Lwt.return
-        (reply reqd ~content_type:"text/html"
-           (Guest_layout.guest_layout ~page_title:"An Error Occured | Mollymawk"
-              ~content:(Error_page.error_layout error)
-              ~icon:"/images/robur.png" ())
-           `Internal_server_error)
+    user_unikernel albatross ~user_name:user.name ~unikernel_name:name
+    >>= fun unikernel_info ->
+    match unikernel_info with
+    | None ->
+        Lwt.return
+          (reply reqd ~content_type:"text/html"
+             (Guest_layout.guest_layout ~page_title:"500 | Mollymawk"
+                ~content:
+                  (Error_page.error_layout
+                     {
+                       code = 500;
+                       success = false;
+                       title = "Error fetching " ^ name;
+                       data =
+                         `String ("An error occured trying to fetch " ^ name);
```

Comment:
Shouldn't that refer to "albatross" to some degree? The "Error fetching " is pretty mysterious.